pkg/tfgen/source.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tfgen
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/common/diag"
+
+	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
+)
+
+// A source of documentation bytes.
+type DocsSource interface {
+	// Get the bytes for a resource with TF token rawname.
+	getResource(
+		rawname string, info tfbridge.ResourceOrDataSourceInfo,
```

Comment:
Repeating the comment before, the signature is very permissive, does it really need to know tfbridge.ResourceOrDataSourceInfo and why?

Reply:
I reduced `info` to `*tfbridge.DocInfo`, which it really does need.